libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -36,21 +60,60 @@ export function toFragment(htmlStrings, ...values) {
   return fragment;
 }
 
-export const getFedsPlaceholderConfig = () => {
-  const { locale } = getConfig();
-  let libOrigin = 'https://milo.adobe.com';
+// TODO we might eventually want to move this to the milo core utilities
+let federatedContentRoot;
+export const getFederatedContentRoot = () => {
+  if (federatedContentRoot) return federatedContentRoot;
+
   const { origin } = window.location;
 
+  federatedContentRoot = allowedOrigins.some((o) => origin.replace('.stage', '') === o)
+    ? origin
+    : 'https://www.adobe.com';
+
   if (origin.includes('localhost') || origin.includes('.hlx.')) {
-    libOrigin = `https://main--milo--adobecom.hlx.${origin.includes('hlx.live') ? 'live' : 'page'}`;
+    federatedContentRoot = `https://main--federal--adobecom.hlx.${origin.includes('hlx.live') ? 'live' : 'page'}`;
   }
 
-  return {
+  return federatedContentRoot;
+};
+
+export const getFederatedUrl = (url = '') => {
+  if (typeof url !== 'string' || !url.includes('/federal/')) return url;
```

Comment:
`/federal/` can be anywhere in the url?  Should we only be matching when it's the first path value?

Reply:
It's always `/federal` or `/locale/federal/` and defined the same way in Akamai, it would make sense to match the Akamai locale aware patterns.

I added a TODO once we port this out of the gnav and make this more of a milo core feature.